_tools/types/main.go

```diff
@@ -0,0 +1,341 @@
+// Generates a report about every driver wich UASTv2 types does it use,
+// both in actual test fixtures and though the source code.
+package main
+
+import (
+	"bufio"
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"path"
+	"reflect"
+	"regexp"
+	"strings"
+	"sync"
+
+	"github.com/bblfsh/sdk/driver/manifest/discovery"
+	"github.com/bblfsh/sdk/uast"
+
+	_ "net/http/pprof"
+)
+
+const (
+	fixtureDir = "fixtures"
+	fixtureExt = ".sem.uast"
+	goDocURL   = "https://godoc.org/github.com/bblfsh/sdk/uast#"
+)
+
+var (
+	reposRootPath = path.Join(".", "drivers")
+	uastNodeRe    = regexp.MustCompile("uast:[^\"]*")
+
+	pprof      = flag.Bool("pprof", false, "start pprof profiler http endpoing")
+	skipUpdate = flag.Bool("skip", false, "skip git clone or pull")
+)
+
+func main() {
+	flag.Parse()
+
+	if *pprof {
+		pprofAddr := "localhost:6060"
+		fmt.Fprintf(os.Stderr, "running pprof on %s\n", pprofAddr)
+		go func() {
+			if err := http.ListenAndServe(pprofAddr, nil); err != nil {
+				log.Fatal("cannot start pprof: %v", err)
+			}
+		}()
+	}
+
+	if err := run(); err != nil {
```

Comment:
We don't seem to rely on any important deferrals here—can we inline `run` in `main`?

(I wrote [ctrl](https://godoc.org/bitbucket.org/creachadair/ctrl) to handle this case for programs that need to do cleanup on exit, but here we don't really seem to need that)

Reply:
Inlining will require pasting `log.Fatal(err)` everywhere, which is okay, but `return`s might be nicer :)

Reply:
Although true, I would prefer keeping it this way for consistency with the other `./_tool/`s

Reply:
> Inlining will require pasting `log.Fatal(err)` everywhere, which is okay, but `return`s might be nicer :)

That's true—though the lint rule about error strings being uncapitalized and unpunctuated makes it ugly logs. Since we're already logging explicitly in a bunch of places, I thought it might be not so bad. To be clear, though, I am not extremely concerned about this, I just brought it up because it stuck out.

Reply:
> Although true, this seems to be a convention that all other `./_tool/`s use, and thus I would prefer keeping it this way for consistency.

Makes sense. I didn't realize we had done this in the others as well. (Maybe I'll import `ctrl` and clean these up later, but it's not important enough to make a mess for)